### PR TITLE
Filter out declaration locations when grouping by definition.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Fixes https://twitter.com/Nick_Craver/status/803754716148604928

### CODE REVIEW

**Comment by CyrusNajmabadi** [comment]
Tagging @dotnet/roslyn-ide

**Comment by CyrusNajmabadi** [comment]
Tagging @MattGertz 

This change improves our FindReferences experience by removing clutter and entries that make it feel like duplicate information is being reported.  This has been reported by customers and internals.

**Comment by MattGertz** [comment]
Approved for RC3.

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit32_prtest please

**Comment by CyrusNajmabadi** [comment]
retest windows_debug_unit32_prtest please

**Comment by DustinCampbell** [comment]
Got a screenshot to share @CyrusNajmabadi?

**Comment by CyrusNajmabadi** [comment]
Retest windows_debug_unit64_prtest  please

**Comment by CyrusNajmabadi** [comment]
Here you go @DustinCampbell 

image

File: src/EditorFeatures/Core/FindReferences/FindReferencesCommandHandler.cs [hunk 1]
```diff
@@ -125,7 +125,8 @@ private IStreamingFindUsagesPresenter GetStreamingPresenter()
                 {
                     // Let the presented know we're starging a search.  It will give us back
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/FindReferences/FindReferencesCommandHandler.cs]: starge!

File: src/VisualStudio/Core/Next/FindReferences/StreamingFindUsagesPresenter.SimpleMessageEntry.cs [hunk 2]
```diff
@@ -28,16 +29,25 @@ private class SimpleMessageEntry : Entry
                 return Task.FromResult<Entry>(referenceEntry);
             }
 
+            private Document TryGetDocument()
+            {
+                return DefinitionBucket.DefinitionItem.SourceSpans.FirstOrDefault().Document;
```
Comment (jasonmalinowski) [inline src/VisualStudio/Core/Next/FindReferences/StreamingFindUsagesPresenter.SimpleMessageEntry.cs]: Did you want a ?. there if the FirstOrDefault() returns null?

File: src/EditorFeatures/Core/Host/IStreamingFindReferencesPresenter.cs [hunk 3]
```diff
@@ -21,7 +21,7 @@ internal interface IStreamingFindUsagesPresenter
         /// search completes <see cref="FindUsagesContext.OnCompletedAsync"/> should be called. 
         /// etc. etc.
         /// </summary>
-        FindUsagesContext StartSearch(string title);
+        FindUsagesContext StartSearch(string title, bool alwaysShowDeclarations);
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Host/IStreamingFindReferencesPresenter.cs]: Can you doc what this does? It's not obvious without doing a lot of digging (or knowing this exact PR exists.)

Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Host/IStreamingFindReferencesPresenter.cs]: k.

**Comment by CyrusNajmabadi** [comment]
@Pilchie Can you review?

**Pull request merged**